src/datachain/model/bbox.py

```diff
@@ -1,47 +1,216 @@
+        return BBox(title=title, coords=list(map(round, coords)))
+
+    def to_voc(self) -> list[int]:
+        """
+        Return the bounding box coordinates in PASCAL VOC format.
+
+        PASCAL VOC format represents bounding boxes as [x_min, y_min, x_max, y_max],
+        where:
+        - (x_min, y_min) are the pixel coordinates of the top-left corner.
+        - (x_max, y_max) are the pixel coordinates of the bottom-right corner.
+
+        Returns:
+            list[int]: The bounding box coordinates in VOC format.
+        """
+        return self.coords
+
+    @staticmethod
+    def from_yolo(
+        coords: Sequence[float],
+        img_size: Sequence[int],
+        title: str = "",
+    ) -> "BBox":
+        """
+        Create a bounding box from YOLO format.
+
+        YOLO format represents bounding boxes as [x_center, y_center, width, height],
+        where:
+        - (x_center, y_center) are the normalized coordinates of the box center.
+        - width and height normalized values define the size of the bounding box.
+
+        Args:
+            coords (Sequence[float]): The bounding box coordinates in YOLO format.
+            img_size (Sequence[int]): The reference image size as `[width, height]`.
+            title (str): The title of the bounding box.
+
+        Returns:
+            BBox: The bounding box data model.
+        """
+        validate_bbox(coords, float)
+        bbox_coords = convert_bbox(coords, img_size, "yolo", "voc")
+        return BBox(title=title, coords=list(map(round, bbox_coords)))
+
+    def to_yolo(self, img_size: Sequence[int]) -> list[float]:
+        """
+        Return the bounding box coordinates in YOLO format.
+
+        YOLO format represents bounding boxes as [x_center, y_center, width, height],
+        where:
+        - (x_center, y_center) are the normalized coordinates of the box center.
+        - width and height normalized values define the size of the bounding box.
+
+        Args:
+            img_size (Sequence[int]): The reference image size as `[width, height]`.
+
+        Returns:
+            list[float]: The bounding box coordinates in YOLO format.
+        """
+        return convert_bbox(self.coords, img_size, "voc", "yolo")
+
+    def point_inside(self, x: int, y: int) -> bool:
+        """
+        Return True if the point is inside the bounding box.
+
+        Assumes that if the point is on the edge of the bounding box,
+        it is considered inside.
+        """
+        x1, y1, x2, y2 = self.coords
+        return x1 <= x <= x2 and y1 <= y <= y2
+
+    def pose_inside(self, pose: Union["Pose", "Pose3D"]) -> bool:
+        """Return True if the pose is inside the bounding box."""
+        return all(
+            self.point_inside(x, y) for x, y in zip(pose.x, pose.y) if x > 0 or y > 0
         )
 
+    @staticmethod
+    def from_list(coords: Sequence[float], title: str = "") -> "BBox":
+        return BBox.from_voc(coords, title=title)
```

Comment:
Not familiar with the logic TBH, but we should probably add a test covering this.

Reply:
Added 🙏